tests/fixture/tmpnet/node.go

```diff
@@ -323,6 +319,59 @@ func (n *Node) GetUniqueID() string {
 	return n.network.UUID + "-" + strings.ToLower(nodeIDString[startIndex:endIndex])
 }
 
+// composeFlags determines the set of flags that should be used to
+// start the node.
+func (n *Node) composeFlags() (FlagsMap, error) {
+	flags := maps.Clone(n.Flags)
+
+	// Apply the network defaults first so that they are not overridden
+	flags.SetDefaults(n.network.DefaultFlags)
+
+	flags.SetDefaults(DefaultTmpnetFlags())
+
+	// Convert the network id to a string to ensure consistency in JSON round-tripping.
+	flags.SetDefault(config.NetworkNameKey, strconv.FormatUint(uint64(n.network.GetNetworkID()), 10))
```

Comment:
nit maybe a bit simpler:

```suggestion
	flags.SetDefault(config.NetworkNameKey, fmt.Sprintf("%d", n.network.GetNetworkID()))
```